### DISCUSSION TITLE
Storing temporary files in /tmp

### DISCUSSION THREAD

**Discussion by rohit8** [vercel/vercel#5320] [Help] [answered]
Is it possible to temporarily store files in /tmp during a function run, so that I can then upload the contents of /tmp to an S3 bucket? What is the limit of the size of the /tmp folder for a given function run?

**Comment by styfle** [comment, marked as answer]
Hi Rohit,

The `/tmp` directory in a Serverless Function has a limit of 512 MB.

However, a better solution is to upload files directly from the client to the S3 bucket using a presigned url.

Then you don't need to proxy the function req body or write temporary files at all.

https://docs.aws.amazon.com/AmazonS3/latest/dev/PresignedUrlUploadObject.html

**Reply by marlonjohnynion** [reply]
@styfle is there a way to overcome that limit (ex: by upgrading..) or is it set for all everyone?

**Reply by styfle** [reply]
There is no way to configure the limit at this time. 

The /tmp storage is ephemeral so it’s best to use some other data store outside of the Serverless Function if you need to reuse files between function invocations.

**Reply by OlypsisAli** [reply]
This is so important. Please add this to the docs. https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage

**Comment by lukasfischer** [comment]
The /tmp hint was very helpful here. I tried to find this on vercel docs, but I couldn't find anything. Highly recommend to add that to https://vercel.com/guides/how-can-i-use-files-in-serverless-functions -- I need it to temporary "save" a pdf file created with pdfkit (obviously not for storing the file, but for tmp saving and loading to send to somwhere else)

**Reply by amyegan** [reply]
Thanks for pointing this out @lukasfischer! Great idea. I'm adding it to our list of docs improvements.

**Reply by amyegan** [reply]
The Docs team updated the Serverless Function files guide! It now includes a section about using temporary storage to save and send a PDF: [vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage)

Thanks again for bringing this up!

**Reply by bhanudatsinhjhala** [reply]
How and where to make /tmp Folder in code and Do we have to make any changess in vercel.json file?? @amyegan

**Reply by amyegan** [reply]
Hi @bhanudatsinhjhala! You do not need to make any changes to `vercel.json` to enable temporary storage. You can set up a serverless function as shown in the [using temporary storage](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage) guide to write to the function's `/tmp` directory and then return the generated file.
